### PR TITLE
Feat/table multiple implementations

### PR DESCRIPTION
**Pull request by KKould**
### What problem does this PR solve?

Abstract SSTable as Table, and implement SkipTable similar to MemTable to support Level 0 memoization

### What is changed and how it works?

- Replace the original SSTable coupling code with Table
- `Config::enable_level_0_memorization` to enable Level 0 memoization

### Code changes

- [x] Has Rust code change

### Check List

Tests 

- [x] Unit test

Side effects

- [x] Performance regression: Consumes more Memory

### Note for reviewer

- At present, Level 0 memorization is only an experimental enableable function, and SSTable is still used by default. 
- The focus of this PR is on the abstraction of Table, which provides the possibility of Table expansion in subsequent scenarios.

### CODE REVIEW

File: src/kernel/lsm/storage.rs [hunk 1]
```diff
@@ -48,7 +49,7 @@ pub(crate) const DEFAULT_MAJOR_THRESHOLD_WITH_SST_SIZE: usize = 10;
 
 pub(crate) const DEFAULT_MAJOR_SELECT_FILE_SIZE: usize = 3;
 
-pub(crate) const DEFAULT_LEVEL_SST_MAGNIFICATION: usize = 10;
+pub(crate) const DEFAULT_LEVEL_SST_MAGNIFICATION: usize = 5;
```
Comment (loloxwg) [inline src/kernel/lsm/storage.rs:53]: Why reduce `DEFAULT_LEVEL_SST_MAGNIFICATION` 
What is the function of this variable?

File: src/kernel/lsm/table/loader.rs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        let inner = Arc::new(ShardingLruCache::new(
+            config.table_cache_size,
+            16,
+            RandomState::default(),
+        )?);
+        let cache = Arc::new(ShardingLruCache::new(
+            config.block_cache_size,
+            16,
+            RandomState::default(),
+        )?);
```
Comment (loloxwg) [inline src/kernel/lsm/table/loader.rs:41]: Do these two caches specify the size of the table and block respectively?

Comment (loloxwg) [reply src/kernel/lsm/table/loader.rs:41]: It is looks like inner cache store the gen which belong to some file

**Review by loloxwg — APPROVED** [review]
lgtm

File: src/kernel/lsm/storage.rs (hunk 1 above)
Comment (KKould) [reply src/kernel/lsm/storage.rs:53]: 这是用于控制每级SSTable数量阈值的数值，DEFAULT_LEVEL_SST_MAGNIFICATION的level次方

Comment (loloxwg) [reply src/kernel/lsm/storage.rs:53]: get

**Pull request merged**